server/src/server.ts

```diff
@@ -58,7 +58,8 @@ connection.onInitialize(async (params: InitializeParams): Promise<InitializeResu
 
   workspaceFolder = params.workspaceFolders?.[0].uri.replace('file://', '')
 
-  const extensionPath = params.initializationOptions.extensionPath as string
+  const extensionPath = params.initializationOptions?.extensionPath ?? workspaceFolder as string

```

Comment:
Thanks a lot for integrating our recommendations in this new iteration!
The code looks sound to properly define some variables in a non-VSCOde environment. However this mix of extension sources and workspace folder looks still a bit eerie. I'd just add a comment here so that we remember what's going on here:
```
// If the language server is not started by VSCode, extensionPath and pathToBitbakeFolder can be undefined
// We provide alternatives here to prevent crashes when these variables are used
```
Otherwise we could also have a look at the code that crashes later and see if we can handle undefined better.
I suggest we keep the current code, with the comment and think about refactoring later when nvim is properly packaged and documented and we get the full picture.

Reply:
```suggestion
  // If the language server is not started by VSCode, extensionPath and pathToBitbakeFolder can be undefined
  // We provide alternatives here to prevent crashes when these variables are used
  const extensionPath = params.initializationOptions?.extensionPath ?? workspaceFolder as string
```

Reply:
Yes, add a comment is a good habit :+1: